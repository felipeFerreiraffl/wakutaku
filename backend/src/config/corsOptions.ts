import cors from "cors";

const origins = ["*"]; // Adicionar a URL da Vercel no final
export const corsOptions: cors.CorsOptions = {
  origin: origins,
  methods: ["GET", "POST", "PUT", "DELETE"],
  allowedHeaders: ["Content-Type", "X-Cache"],
  credentials: false,
  optionsSuccessStatus: 200,
};
