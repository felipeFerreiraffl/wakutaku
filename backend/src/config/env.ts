import { envVar } from "../types/env.js";

// Vari√°veis de ambiente
export const env = {
  NODE_ENV: envVar.NODE_ENV,
  PORT: envVar.PORT,
  API_URL: envVar.API_URL,
  REDIS_URL: envVar.REDIS_URL,
  REDIS_USERNAME: envVar.REDIS_USERNAME,
  REDIS_PASSWORD: envVar.REDIS_PASSWORD,
};
