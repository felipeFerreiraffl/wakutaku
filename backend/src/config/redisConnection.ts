import type { NextFunction, Request, Response } from "express";
import { BasicClientSideCache, createClient } from "redis";
import {
  setSuccessMessage,
  type JikanError,
} from "../middlewares/statusHandler.js";

// Configuração do cache
const cacheConfig = new BasicClientSideCache({
  ttl: 0, // Não expira
  maxEntries: 0, // Quantidade máxima de entradas para guardar
  evictPolicy: "FIFO", // Padrão de expurgação de dados do cache (FIFO = First In, First Out)
});

const client = createClient({
  RESP: 3,
  clientSideCache: cacheConfig,
});

// Sucesso
client.on("connection", (res: Response) => {
  const response = res.json({ message: "Conexão feita com sucesso" });
  setSuccessMessage(res, { response });
});

// Erro no Redis
client.on(
  "error",
  (err: JikanError, req: Request, res: Response, next: NextFunction) => {
    console.error(`Erro ao conectar o banco de dados Redis: ${err.message}`);
    res.status(err.status ?? 500).json({ success: false, error: err.message });
  }
);

await client.connect();

await client.set("key", "value");
const value = client.get("key");
console.log(`Valor do Redis: ${value}`);

// await client.expire("key", 60);
