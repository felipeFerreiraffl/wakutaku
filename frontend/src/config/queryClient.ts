import { QueryClient } from "@tanstack/react-query";

// Confiração inicial do TanStack Query
export const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: 2 * 60 * 1000, // Tempo padrão de cache
      gcTime: 10 * 60 * 1000,
      retry: 1,
      refetchOnWindowFocus: false,
    },
  },
});
