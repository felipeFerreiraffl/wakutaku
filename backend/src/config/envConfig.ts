import { envVars } from "../types/env.js";

export const env = {
  NODE_ENV: envVars.NODE_ENV,
  API_URL: envVars.API_URL,
  PORT: envVars.PORT,
  REDIS_URL: envVars.REDIS_URL,
  REDIS_USERNAME: envVars.REDIS_USERNAME,
  REDIS_PASSWORD: envVars.REDIS_PASSWORD,
};
