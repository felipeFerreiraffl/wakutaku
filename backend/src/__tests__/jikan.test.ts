import { describe, expect, it, vi } from "vitest";
import { envVar } from "../config/envConfig";

vi.stubGlobal(
  "fetch",
  vi.fn(() => {})
);

describe("GET /season_stats", () => {
  it("retorna as estatísticas da temporada atual de animes", async () => {
    const response = await fetch(
      `http://localhost:${envVar.PORT}/api/season_stats`
    );

    await expect(response.json()).resolves.toEqual({
      success: true,
      data: {
        totalCount: 200,
        frequentGenre: "Action",
        frequentDemography: "Shounen",
        averageScore: 10,
      },
    });
  });

  it.skip("retorna TOO_MANY_REQUESTS quando há muitas requisições", async () => {});
});
