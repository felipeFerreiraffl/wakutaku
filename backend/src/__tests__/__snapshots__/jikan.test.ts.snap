// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testes de rotas da JIKAN > GET /season_stats > retorna as estat√≠sticas da temporada atual de animes 1`] = `
{
  "averageScore": Any<Number>,
  "frequentDemography": Any<String>,
  "frequentGenre": Any<String>,
  "totalCount": Any<Number>,
}
`;
